src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            BulldozerProducer<T> bulldozerProducer = new BulldozerProducer<>();
+            return (List<T>) bulldozerProducer.get();
+        } else if (type.equals(Excavator.class)) {
+            ExcavatorProducer<T> excavatorProducer = new ExcavatorProducer<>();
+            return (List<T>) excavatorProducer.get();
+        } else if (type.equals(Truck.class)) {
+            TruckProducer<T> truckProducer = new TruckProducer<>();
+            return (List<T>) truckProducer.get();
+        }
+        return new ArrayList<>();
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/pecs/pecs.html#consider-creating-a-local-variable-in-your-implementation-of-getallclass-type-method

Reply:
My bad